### PR TITLE
test python 3.10

### PR DESCRIPTION
**Pull request by saolsen**
Adds python 3.10 as a build and test target.
Removes musl specific wheels, python wheels now work for both glibc and musl based distros.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -593,7 +567,7 @@ jobs:
     runs-on: macos-10.15
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
```
Comment (samscott89) [inline .github/workflows/release.yml:570]: lol yaml

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -564,7 +538,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: ["3.6", "3.7", "3.8", "3.9"]
```
Comment (samscott89) [inline .github/workflows/release.yml:541]:
```suggestion
        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
```

**Review by samscott89 — COMMENTED** [review]
Take it or leave it: was looking at the changes between 1.6 and 2.2 for cibuildwheel. Just checking if we care about any of the following:

- Dropped support for 3.5 (don't care)
- Can configure cibuildwheel through [config file instead](https://cibuildwheel.readthedocs.io/en/stable/options/#setting-options) of env variables -- worth doing?
- It now support `musllinux` I didn't realise that was a thing... anything to do there? https://cibuildwheel.readthedocs.io/en/stable/changelog/#v220 we might currently filter them out accidentally
- Seems like there is a GitHub action we could be using directly instead: https://cibuildwheel.readthedocs.io/en/stable/setup/#github-actions

File: .github/workflows/release.yml (hunk 1 above)
Comment (saolsen) [reply .github/workflows/release.yml:570]: yah

**Comment by saolsen** [comment]
> * Can configure cibuildwheel through [config file instead](https://cibuildwheel.readthedocs.io/en/stable/options/#setting-options) of env variables -- worth doing?

meh, maybe later

> * It now support `musllinux` I didn't realise that was a thing... anything to do there? https://cibuildwheel.readthedocs.io/en/stable/changelog/#v220 we might currently filter them out accidentally

These work, I see them in there.

> * Seems like there is a GitHub action we could be using directly instead: https://cibuildwheel.readthedocs.io/en/stable/setup/#github-actions

meh, maybe later

**Review by samscott89 — APPROVED** [review]

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -16,7 +16,7 @@ jobs:
       - name: Set version env
         run: echo "oso_version=$(cat VERSION)" >> $GITHUB_ENV
       - name: Check github ref matches
-        if: github.ref != 'refs/heads/main'
+        if: github.ref != 'refs/heads/main
```
Comment (samscott89) [inline .github/workflows/release.yml:19]:
```suggestion
        if: github.ref != 'refs/heads/main'
```

**Pull request merged**